### ISSUE TITLE
Hyperparameter range issues

### ISSUE THREAD

**Issue by wbenoit26** [ML4GW/BBHNet#307]
As I try to do some basic hyperparameter searches, I've found some edge cases that we might want to handle, or at least raise an error message for:

- A bad combination of `downweight`, `injection_prob`, and `glitch_prob` can drive [this](https://github.com/ML4GW/BBHNet/blob/3e9150cb8bbf5a10e71ac8fe2aae9a3d7ad8949d/projects/sandbox/train/train/data_structures.py#L90) `prob` greater than 1.
- If the glitch dataset has a significant imbalance between Hanford and Livingston glitches, then too high a `glitch_prob` can make [`num_coinc`](https://github.com/ML4GW/BBHNet/blob/3e9150cb8bbf5a10e71ac8fe2aae9a3d7ad8949d/projects/sandbox/train/train/validation.py#L347) greater than the number of glitches in one of the ifo sets.
- Possibly related to the glitch dataset imbalance, I've had cases where [splitting](https://github.com/ML4GW/BBHNet/blob/3e9150cb8bbf5a10e71ac8fe2aae9a3d7ad8949d/projects/sandbox/train/train/validation.py#L349) the glitches creates tensors of different sizes due to floating point errors. For example, in my case, `num_coinc = 16`, `num_h1 = 18`, and `num_l1 = 196`. When we `split`, `int((16/18) * 18) = 16`, but `int((16/196) * 196) = 15`.

**Comment by alecgunny** [comment]
> A bad combination of downweight, injection_prob, and glitch_prob can drive [this](https://github.com/ML4GW/BBHNet/blob/3e9150cb8bbf5a10e71ac8fe2aae9a3d7ad8949d/projects/sandbox/train/train/data_structures.py#L90) prob greater than 1.
This is a good catch and there should definitely be an explicit `0 <= prob <= 1` check. During HP sampling you could also maybe sample `glitch_prob` and `injection_prob` first and then sample `downweight` with a min value of `1 - (1 - injection_prob**0.5) / glitch_prob` (max'd with 0 obviously)

For the glitch errors, we should probably do away with doing anything special with glitches during validation at all. I suspect a fair number of the ones we use now are actually from the training set: when you run with downweighted glitches, you get basically 100% recall up at 90% glitch specificity almost immediately. And as I note in #302 , if glitches in the validation set are actually the problem, then they'll bubble up in the relevant metrics anyway.

I would like to do one last validation overhaul, one that more closely mimics the validation in the example notebook and maybe does some light time shifting, so this will be a good opportunity to do away with problems. For now, if you want to just delete the glitch loader altogether, I would be more than fine with that.

**Comment by wbenoit26** [comment]
Yeah, that range restriction is what I'm doing currently. I've also done a couple quick workarounds to deal with the glitch issues, but I agree, a fuller treatment is needed.
